### ISSUE TITLE
RepoAuthoritative: HHVM looks for array_map() in the current namespace

### ISSUE THREAD

**Issue by edwardspec** [facebook/hhvm#7572]
Testcase is attached.
The error only occurs in RepoAuthoritative mode.

### HHVM Version

HipHop VM 3.17.0 (rel)
Compiler: tags/HHVM-3.17.0-0-ga34af693b558ed98ffafc3f5127e00959e145e4f
Repo schema: 8eda451fe80742a18e0e2e8917aa4053c2c1afe7

### Standalone code, or other way to reproduce the problem

```
<?php

namespace WhateverNamespace;

class WhateverTest {
	public function filter( $array ) {
		$mapFunction = function ( $item ) {
			return "whatever";
		};

		return array_map( $mapFunction, $array );
	}
}

$t = new WhateverTest();
$ret = $t->filter( array(  "lol", "wow", "doge" ) );

var_dump( $ret );
```

### Expected result

array(3) {
  [0]=>
  string(8) "whatever"
  [1]=>
  string(8) "whatever"
  [2]=>
  string(8) "whatever"
}

### Actual result

\nFatal error: Call to undefined function WhateverNamespace\\array_map() in /home/mediawiki/w/testcase.php on line 11

### Impact

MediaWiki 1.27 with Extension:MobileFrontend can't be run in RepoAuthoritative mode.

**Comment by mofarrell** [comment]
Compiling the repo with `-vElideAutoloadInvokes=false` allows this to compile.  It seems that a couple of our builtins (the ones written in hhas for performance) are not known about during the hhbbc optimization phase.  This means the optimization provided by eliding autoload invocations can break the use of those functions inside of namespaces.

**Comment by mofarrell** [comment]
@edwardspec It should be fixed on master now, and should be fixed in 3.17.1
Comment here if it is still an issue.
